src/steps/access.ts

```diff
@@ -1,105 +1,114 @@
 import {
   createDirectRelationship,
   createIntegrationEntity,
-  Entity,
   IntegrationStep,
   IntegrationStepExecutionContext,
   RelationshipClass,
-  IntegrationMissingKeyError,
+  getTime,
 } from '@jupiterone/integration-sdk-core';
 
-import { createAPIClient } from '../client';
+import { createAPIClient } from '../provider/client';
 import { IntegrationConfig } from '../types';
-import { ACCOUNT_ENTITY_KEY } from './account';
 
 export async function fetchUsers({
   instance,
   jobState,
 }: IntegrationStepExecutionContext<IntegrationConfig>) {
   const apiClient = createAPIClient(instance.config);
 
-  const accountEntity = (await jobState.getData(ACCOUNT_ENTITY_KEY)) as Entity;
-
-  await apiClient.iterateUsers(async (user) => {
-    const userEntity = createIntegrationEntity({
-      entityData: {
-        source: user,
-        assign: {
-          _type: 'acme_user',
-          _class: 'User',
-          username: 'testusername',
-          email: 'test@test.com',
-          // This is a custom property that is not a part of the data model class
-          // hierarchy. See: https://github.com/JupiterOne/data-model/blob/master/src/schemas/User.json
-          firstName: 'John',
+  await apiClient.getUsers(async (data) => {
+    await jobState.addEntity(
+      createIntegrationEntity({
+        entityData: {
+          source: data,
+          assign: {
+            _type: 'fastly_user',
+            _class: 'User',
+            _key: `fastly-user:${data.id}`,
+            id: data.id,
+            displayName: data.name || data.id,
+            name: data.name,
+            username: data.login,
+            login: data.login,
+            role: data.role,
+            mfaEnabled: data.two_factor_auth_enabled,
+            mfaSetupRequired: data.two_factor_setup_required,
+            locked: data.locked,
+            active: !(data.deleted_at || data.locked),
+            createdOn: getTime(data.created_at),
+            updatedOn: getTime(data.updated_at),
+            deletedOn: getTime(data.deleted_at),
+          },
         },
-      },
-    });
+      }),
+    );
 
-    await Promise.all([
-      jobState.addEntity(userEntity),
-      jobState.addRelationship(
-        createDirectRelationship({
-          _class: RelationshipClass.HAS,
-          from: accountEntity,
-          to: userEntity,
-        }),
-      ),
-    ]);
+    await jobState.addRelationship(
+      createDirectRelationship({
+        _class: RelationshipClass.HAS,
+        fromKey: `fastly-account:${instance.config.customerId}`,
+        fromType: 'fastly_account',
+        toKey: `fastly-user:${data.id}`,
+        toType: 'fastly_user',
+      }),
```

Comment:
It would be better to create this using the user entity and account entity, instead of using this form of the function. This form really only exists as a legacy mechanism for older integrations where it would be significant work to up-fit to the form that takes the from/to entities. This helps avoid mistakes in the relationship construction where a source or target entity does not actually exist in the job state.

See https://github.com/JupiterOne/sdk/blob/40268ff5b950a468bbc4538925d90d7c47f1727a/packages/integration-sdk/CHANGELOG.md#290---2020-08-05 for examples of how to make it easy.

The `Account` entity can be found by `await jobState.findEntity(known_key)`. Of course, the step that builds the `Account` entity must precede those that look for it in the `jobState`.